x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/all_exception_items_table/index.tsx

```diff
@@ -0,0 +1,396 @@
+
+import { ExceptionsViewerPagination } from './pagination';
+import { ExceptionsViewerUtility } from './utility_bar';
+import { ExceptionsViewerItems } from './all_items';
+import { EditExceptionFlyout } from '../edit_exception_flyout';
+import { AddExceptionFlyout } from '../add_exception_flyout';
+import * as i18n from './translations';
+import { useFindExceptionListReferences } from '../../logic/use_find_references';
+import type { Rule } from '../../../../detections/containers/detection_engine/rules/types';
+
+const STATES_SEARCH_HIDDEN: ViewerState[] = ['error', 'empty'];
+const STATES_PAGINATION_UTILITY_HIDDEN: ViewerState[] = [
+  'loading',
+  'empty_search',
+  'empty',
+  'error',
+  'searching',
+];
+
+const initialState: State = {
+  pagination: {
+    pageIndex: 0,
+    pageSize: 25,
+    totalItemCount: 0,
+    pageSizeOptions: [1, 5, 10, 25, 50, 100, 200, 300],
+  },
+  exceptions: [],
+  exceptionToEdit: null,
+  currenFlyout: null,
+  viewerState: 'loading',
+};
+
+export interface GetExceptionItemProps {
+  pagination?: Partial<Pagination>;
+  search?: string;
+  filters?: string;
+}
+
+interface ExceptionsViewerProps {
+  rule: Rule | null;
+  listType: ExceptionListTypeEnum;
+  onRuleChange?: () => void;
+}
+
+const ExceptionsViewerComponent = ({
+  rule,
+  listType,
+  onRuleChange,
+}: ExceptionsViewerProps): JSX.Element => {
+  const { services } = useKibana();
+  const toasts = useToasts();
+  const [{ canUserCRUD, hasIndexWrite }] = useUserData();
+  const [isReadOnly, setReadOnly] = useState(true);
+  const [lastUpdated, setLastUpdated] = useState<null | string | number>(null);
+  const exceptionListsToQuery = useMemo(
+    () =>
+      rule != null && rule.exceptions_list != null
+        ? rule.exceptions_list.filter((list) => list.type === listType)
+        : [],
+    [listType, rule]
+  );
+
+  // Reducer state
+  const [{ exceptions, pagination, currenFlyout, exceptionToEdit, viewerState }, dispatch] =
+    useReducer(allExceptionItemsReducer(), {
+      ...initialState,
+    });
+
+  // Reducer actions
+  const setExceptions = useCallback(
+    ({
+      exceptions: newExceptions,
+      pagination: newPagination,
+    }: UseExceptionListItemsSuccess): void => {
+      setLastUpdated(Date.now());
+
+      dispatch({
+        type: 'setExceptions',
+        exceptions: newExceptions,
+        pagination: newPagination,
+      });
+    },
+    [dispatch]
+  );
+
+  const setViewerState = useCallback(
+    (state: ViewerState): void => {
+      dispatch({
+        type: 'setViewerState',
+        state,
+      });
+    },
+    [dispatch]
+  );
+
+  const setFlyoutType = useCallback(
+    (flyoutType: ViewerFlyoutName): void => {
+      dispatch({
+        type: 'updateFlyoutOpen',
+        flyoutType,
+      });
+    },
+    [dispatch]
+  );
+
+  const [_, allReferences] = useFindExceptionListReferences(exceptionListsToQuery);
+
+  const handleFetchItems = useCallback(
```

Comment:
Per review with @WafaaNasr - move API logic to custom hook that returns this method for us to use.

Reply:
Going to be done as follow up - likely with Waafa's upcoming work on the list detail page.